types/simulation/types.go

```diff
@@ -75,7 +76,16 @@ func NewOperationMsgBasic(route, name, comment string, ok bool, msg []byte) Oper
 }
 
 // NewOperationMsg - create a new operation message from sdk.Msg
-func NewOperationMsg(msg sdk.Msg, ok bool, comment string) OperationMsg {
+func NewOperationMsg(msg sdk.Msg, ok bool, comment string, cdc *codec.ProtoCodec) OperationMsg {
+	if reflect.TypeOf(msg) == reflect.TypeOf(sdk.ServiceMsg{}) {
+		srvMsg, ok := msg.(sdk.ServiceMsg)
+		if !ok {
+			panic("failed: type assert")
+		}
+		bz := cdc.MustMarshalJSON(srvMsg.Request)
+
+		return NewOperationMsgBasic(msg.Route(), msg.Type(), comment, ok, bz)
```

Comment:
So we shouldn't worry about Route and Msg here I think. Instead we should use the MethodName.